mutablesecurity/solutions/implementations/clamav/code.py

```diff
@@ -0,0 +1,752 @@
+
+class InstalledVersion(BaseInformation):
+    class InstalledVersionFact(FactBase):
+        command = (
+            "apt-cache policy clamav | grep -i Installed | cut -d ' ' -f 4"
+        )
+
+        @staticmethod
+        def process(output: typing.List[str]) -> str:
+            return output[0]
+
+    IDENTIFIER = "version"
+    DESCRIPTION = "Installed version"
+    INFO_TYPE = StringDataType
+    PROPERTIES = [
+        InformationProperties.METRIC,
+        InformationProperties.READ_ONLY,
+    ]
+    DEFAULT_VALUE = None
+    GETTER = InstalledVersionFact
+    SETTER = None
+
+
+class TotalInfectedFiles(BaseInformation):
+    class TotalInfectedFilesFact(FactBase):
+        @staticmethod
+        def command() -> str:
+            return (
+                f"cat {ScanLogLocation.get()}| grep -oP 'Infected"
+                " files:\s+\K\w+' | awk '{T+= $NF} END { print T }'"
+            )
+
+        @staticmethod
+        def process(output: typing.List[str]) -> int:
+            return int(output[0])
+
+    IDENTIFIER = "total_infected_files_detected"
+    DESCRIPTION = "Total number of infected files detected overall."
+    INFO_TYPE = IntegerDataType
+    PROPERTIES = [
+        InformationProperties.METRIC,
+        InformationProperties.READ_ONLY,
+    ]
+    DEFAULT_VALUE = None
+    GETTER = TotalInfectedFilesFact
+    SETTER = None
+
+
+class DailyInfectedFiles(BaseInformation):
+    class DailyInfectedFilesCountFact(FactBase):
+        @staticmethod
+        def command() -> str:
+            current_date = datetime.today().strftime("%Y:%m:%d")
+
+            return (
+                f"cat {ScanLogLocation.get()}| grep -B 6 'Start Date:"
+                f" {current_date}' | grep 'Infected files' | egrep -o '[0-9]+'"
+            )
+
+        @staticmethod
+        def process(output: typing.List[str]) -> int:
+            temp = [int(elem) for elem in output]
+            return sum(temp)
+
+    IDENTIFIER = "daily_infected_files_detected"
+    DESCRIPTION = "Total number of infected files detected today."
+    INFO_TYPE = IntegerDataType
+    PROPERTIES = [
+        InformationProperties.METRIC,
+        InformationProperties.READ_ONLY,
+    ]
+    DEFAULT_VALUE = None
+    GETTER = DailyInfectedFilesCountFact
+    SETTER = None
+
+
+class UbuntuRequirement(BaseTest):
+    IDENTIFIER = "ubuntu"
+    DESCRIPTION = "Checks if the operating system is Ubuntu."
+    TEST_TYPE = TestType.REQUIREMENT
+    FACT = CheckIfUbuntu
+
+
+class ActiveClamAVDatabase(BaseTest):
+    IDENTIFIER = "active_database"
+    DESCRIPTION = (
+        "Checks if the ClamAV virus database service is active. Freshclam is a"
+        " virus database update tool for ClamAV."
+    )
+    TEST_TYPE = TestType.OPERATIONAL
+    FACT = ActiveService
+    FACT_ARGS = ("clamav-freshclam",)
+
+
+class ActiveClamAVDaemon(BaseTest):
+    IDENTIFIER = "active_daemon"
+    DESCRIPTION = (
+        "Checks if the ClamAV daemon service is active. Daemon checks"
```

Comment:
Same as for Freshclam. Can we mention this in the main description?

Reply:
Yep, moved everything into the description.